### ISSUE TITLE
Performace degradation after auto-vectorization

### ISSUE THREAD

**Issue by hhy3** [facebookresearch/faiss#2748]
# Summary

# Platform

OS:  5.10.102.1-microsoft-standard-WSL2 

Faiss version:  6d0294ba

Installed from:  compiled by myself

Faiss compilation options:  

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [x] Python

# Reproduction instructions

 
The performance has decreased since #2716 , seems that my compiler ignores the `_GCC_` flag and doesn't auto vectorize correctly

**Comment by mdouze** [comment]
Thanks for flagging.

**Comment by alexanderguzhva** [comment]
What is the compiler version that you're using?

**Comment by alexanderguzhva** [comment]
Also, could you please check whether you're using AVX2 version of Faiss.

**Comment by hhy3** [comment]
> What is the compiler version that you're using?

g++ 10.3.0

**Comment by hhy3** [comment]
> Also, could you please check whether you're using AVX2 version of Faiss.

Yep, I've been using avx2 before. It seems to be having some problems today after pulling the main branch and compiling

**Comment by alexanderguzhva** [comment]
I will land the fix shortly and will let you know. It is notorious case of detecting the proper compiler, because it was discovered that sometimes a compiler can do better optimizations than a hand-coded kernel.

**Comment by alexanderguzhva** [comment]
[#2750](https://github.com/facebookresearch/faiss/pull/2750) should fix the problem, and it is being deployed. Please let know its effect of the performance on your end (still slower, runs as fast as before, became faster) after it gets deployed. Thanks.

**Comment by hhy3** [comment]
> [#2750](https://github.com/facebookresearch/faiss/pull/2750) should fix the problem, and it is being deployed. Please let know its effect of the performance on your end (still slower, runs as fast as before, became faster) after it gets deployed. Thanks.

Thanks for your fix! Now the performance runs as fast as before.
